pywinauto/controls/uia_controls.py

```diff
@@ -848,3 +848,108 @@ def item_by_path(self, path, exact=False):
             raise IndexError()
 
         return menu
+
+
+# ====================================================================
+class TooltipWrapper(uiawrapper.UIAWrapper):
+
+    """Wrap an UIA-compatible Tooltip control"""
+
+    _control_types = [
+        IUIA().UIA_dll.UIA_ToolTipControlTypeId
+    ]
+
+    # -----------------------------------------------------------
+    def __init__(self, elem):
+        """Initialize the control"""
+        super(TooltipWrapper, self).__init__(elem)
+
+
+# ====================================================================
+class ToolbarWrapper(uiawrapper.UIAWrapper):
+
+    """Wrap an UIA-compatible ToolBar control
+
+    The control's children usually are: Buttons, SplitButton,
+    MenuItems, ThumbControls, TextControls, Separators, CheckBoxes.
+    Notice that ToolTip controls are children of the top window and
+    not of the toolbar.
+    """
+
+    _control_types = [
+        IUIA().UIA_dll.UIA_ToolBarControlTypeId
+    ]
+
+    # -----------------------------------------------------------
+    def __init__(self, elem):
+        """Initialize the control"""
+        super(ToolbarWrapper, self).__init__(elem)
+
+    @property
+    def writable_props(self):
+        """Extend default properties list."""
+        props = super(ToolbarWrapper, self).writable_props
+        props.extend(['button_count'])
+        return props
+
+    # ----------------------------------------------------------------
+    def texts(self):
+        """Return texts of the Toolbar"""
+        return self.children_texts()
+
+    #----------------------------------------------------------------
+    def button_count(self):
+        """Return a number of buttons on the ToolBar"""
+        return len(self.children())
+
+    # ----------------------------------------------------------------
+    def button(self, button_identifier, exact=True, by_tooltip=False):
+        """Return the button at index button_index"""
+
+        cc = self.children()
+        texts = [c.window_text() for c in cc]
+        if isinstance(button_identifier, six.string_types):
+            self.actions.log('Toolbar buttons: ' + str(texts))
+
+            if by_tooltip:
+                texts = self.tip_texts()
```

Comment:
I missed that one with "copy-paste". However, I think it has to be removed as I can't see a clean method in UIA for retrieving tooltips. So far only hovering mouse over the button triggers drawing the tooltip window.